cmd/node/factory/structs.go

```diff
@@ -2198,13 +2202,20 @@ func newValidatorStatisticsProcessor(
 	initialNodes := processComponents.nodesConfig.InitialNodes
 	storageService := processComponents.data.Store
 
+	var peerDataPool peer.DataPool
```

Comment:
to avoid the if else you could do something like:
``` go
peerDataPool := processComponents.data.MetaDatapool
if processComponents.shardCoordinator.SelfId() < processComponents.shardCoordinator.NumberOfShards() {
		peerDataPool = processComponents.data.Datapool
}
```

Reply:
Done, but the `var peerDataPool peer.DataPool` needs to stay since `DataPool` and `MetaDatapool` are different types:

```go
var peerDataPool peer.DataPool =  processComponents.data.MetaDatapool
if processComponents.shardCoordinator.SelfId() < processComponents.shardCoordinator.NumberOfShards() {
    peerDataPool = processComponents.data.Datapool
}
```